### PR TITLE
More robust i18n try3

### PR DESCRIPTION
**Pull request by godiard**
Replaces https://github.com/sugarlabs/sugar/pull/557

### CODE REVIEW

**Comment by godiard** [comment]
@samdroid-apps, I addressed your comments in the next patch (Before open....)

File: extensions/cpsection/language/model.py [hunk 1]
```diff
@@ -81,49 +81,54 @@ def _initialize():
...
+
+    new_lang_line = 'LANG="%s"\n' % lang_env
+    new_language_line = 'LANGUAGE="%s"\n' % language_env
+
+    if lang_line == new_lang_line and language_line == new_language_line:
+        return
 
+    try:
+        with open(path, 'w') as fd:
+            fd.write(new_lang_line)
```
Comment (quozl) [inline extensions/cpsection/language/model.py]: Lines 105 to 110 do not preserve previous unrelated contents of file.  Since the file can contain locale environment variables like currency and date format, this is impolite to other software. :smile:

File: extensions/cpsection/language/model.py [hunk 2]
```diff
@@ -81,49 +81,54 @@ def _initialize():
...
+    if 'LANGUAGE' in os.environ:
+        lang = os.environ['LANGUAGE']
+        lang = lang.strip()
+        if lang.endswith('UTF-8'):
+            lang = lang.replace('UTF-8', 'utf8')
+        langlist = lang.split(':')
+
+    if 'LANG' in os.environ:
+        lang = os.environ['LANG']
+        lang = lang.strip()
```
Comment (quozl) [inline extensions/cpsection/language/model.py]: Lines 127 to 131 seem remarkably similar to lines 120 to 124 (not 125), so perhaps a local function inside get_languages would ease reading.

File: extensions/cpsection/language/model.py [hunk 3]
```diff
@@ -81,49 +81,54 @@ def _initialize():
...
 
+    try:
+        with open(path, 'w') as fd:
+            fd.write(new_lang_line)
+            fd.write(new_language_line)
+            fd.flush()
+            # be sure all is flushed
+            os.fsync(fd)
+    except:
+        logging.exception('Error writing .i18n file')
```
Comment (quozl) [inline extensions/cpsection/language/model.py]: Line 112 logs an exception only.  There's no UI response, so the user will think the change will be effective, but it won't be.  Why not catch the exception in caller and revert the change?

**Comment by godiard** [comment]
Replaced by https://github.com/sugarlabs/sugar/pull/560
